packages/contracts-periphery/src/config/drippie.ts

```diff
@@ -0,0 +1,108 @@
+import assert from 'assert'
+
+import { BigNumberish, ethers } from 'ethers'
+import { Interface } from 'ethers/lib/utils'
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+
+import { Etherscan } from '../etherscan'
+
+export interface DripConfig {
+  interval: BigNumberish
+  dripcheck: string
+  checkparams?: any
+  actions: Array<{
+    target: string
+    value?: BigNumberish
+    data?:
+      | string
+      | {
+          fn: string
+          args: any[]
+        }
+  }>
+}
+
+export interface DrippieConfig {
+  [name: string]: DripConfig
+}
+
+export const getDrippieConfig = async (
+  hre: HardhatRuntimeEnvironment
+): Promise<Required<DrippieConfig>> => {
+  let config: DrippieConfig
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    config = require(`../../config/drippie/${hre.network.name}.ts`).default
+  } catch (err) {
+    throw new Error(
+      `error while loading drippie config for network: ${hre.network.name}, ${err}`
+    )
+  }
+
+  return parseDrippieConfig(hre, config)
+}
+
+export const encodeDripCheckParams = (
+  iface: Interface,
+  params: any
+): string => {
+  return ethers.utils.defaultAbiCoder.encode(
```

Comment:
Maybe this could be in a try catch that says make sure to have an event with the correct name